api.go

```diff
@@ -0,0 +1,176 @@
+package httpapi
+
+import (
+	"fmt"
+	"io/ioutil"
+	gohttp "net/http"
+	"os"
+	"path"
+	"strings"
+
+	iface "github.com/ipfs/interface-go-ipfs-core"
+	caopts "github.com/ipfs/interface-go-ipfs-core/options"
+	homedir "github.com/mitchellh/go-homedir"
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr-net"
+)
+
+const (
+	DefaultPathName = ".ipfs"
+	DefaultPathRoot = "~/" + DefaultPathName
+	DefaultApiFile  = "api"
+	EnvDir          = "IPFS_PATH"
+)
+
+type HttpApi struct {
+	url     string
+	httpcli gohttp.Client
+
+	applyGlobal func(*RequestBuilder)
+}
+
+//TODO: Return errors here
+func NewLocalApi() iface.CoreAPI {
+	baseDir := os.Getenv(EnvDir)
+	if baseDir == "" {
+		baseDir = DefaultPathRoot
+	}
+
+	return NewPathApi(baseDir)
+}
+
+func NewPathApi(p string) iface.CoreAPI {
+	a := ApiAddr(p)
+	if a == nil {
+		return nil
+	}
+	return NewApi(a)
+}
+
+func ApiAddr(ipfspath string) ma.Multiaddr {
```

Comment:
Should this return an error if we get an IO error other than one that matches `os.IsNotExist`?